generate.py

```diff
@@ -107,11 +106,7 @@ def main(
 
     fabric = L.Fabric(accelerator=accelerator, devices=1)
 
-    if dtype is not None:
-        dt = getattr(torch, dtype, None)
-        if not isinstance(dt, torch.dtype):
-            raise ValueError(f"{dtype} is not a valid dtype.")
-        dtype = dt
+    dtype = torch.bfloat16 if torch.cuda.is_bf16_supported() else torch.float32
```

Comment:
Would `torch.float16` be a better option for people running on V100?

The reasoning is that if you don't have `bfloat16`, you likely also don't have a lot of memory.

Reply:
float16 seems to deteriorate the quality of results too much.